### PR TITLE
feat: Some GUI improvements in re: Part Instances rewrite

### PR DESCRIPTION
**Pull request by jstarpl**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

A key change is caching of the temporary PartInstances in the RundownTiming component, so that they aren't regenerated every timing calculation. Also, subscriptions have been re-arranged so that there's fewer of them, combined into larger ones, hopefully helping with the GUI-related server load and increasing responsiveness. Also, subscriptions have been modified to use as much of the available indexes as possible.

There is still a known issue where, when clearing an infinite, the infinite preview in the rundown will not be perfectly stable. While having that issue in mind, I would like to merge this into R24 already, to test the improvements made so far.

* **Other information**:

**Status**

- [ ] The functionality has been tested by the PR author
- [ ] The functionality has been tested by NRK

### CODE REVIEW

File: meteor/client/ui/RundownView.tsx [hunk 1]
```diff
@@ -1504,7 +1504,12 @@ export const RundownView = translateWithTracker<IProps, IState, ITrackedProps>((
 				playlistId,
 			})
 			this.autorun(() => {
-				let playlist = RundownPlaylists.findOne(playlistId)
+				let playlist = RundownPlaylists.findOne(playlistId, {
```
Comment (nytamin) [inline meteor/client/ui/RundownView.tsx:1507]: To ensure type safety, do a Pick<RundownPlaylist, '_id' | 'studioId'> here

File: meteor/client/lib/MeteorReactComponent.ts [hunk 2]
```diff
@@ -49,8 +49,9 @@ export class MeteorReactComponent<IProps, IState = {}> extends React.Component<I
 		return computation
 	}
 	subscriptionsReady(): boolean {
-		for (let sub of Object.values(this._subscriptions)) {
-			if (!sub.ready()) {
+		const values = Object.values(this._subscriptions)
```
Comment (nytamin) [inline meteor/client/lib/MeteorReactComponent.ts:52]: I'd prefer variable naming to be more descriptive, eg "subscriptions" is better than "values" imo.

File: meteor/client/ui/RundownView.tsx [hunk 3]
```diff
@@ -1557,6 +1570,84 @@ export const RundownView = translateWithTracker<IProps, IState, ITrackedProps>((
...
 				}
 			})
+			this.autorun(() => {
+				let playlist = RundownPlaylists.findOne(playlistId, {
+					fields: {
+						_id: 1,
+					},
+				})
+				if (playlist) {
+					const rundownIds = playlist.getRundownIDs(undefined, {
```
Comment (nytamin) [inline meteor/client/ui/RundownView.tsx:1580]: Should this be refactored into a `playlist.getRundownUnorderedIDs()` function?

**Review by nytamin — COMMENTED** [review]
Looks good in general, the limiting of reactivity looks good when reading it through.

I added a few comments on adding `Pick` or `Omit` types in some cases where `fields` modifiers are used, there are a few more cases they could be added.

File: meteor/client/ui/RundownView.tsx (hunk 3 above)
Comment (jstarpl) [reply meteor/client/ui/RundownView.tsx]: Refactored into a general `getRundownUnorderedIDs()` function.

File: meteor/client/ui/RundownView.tsx (hunk 1 above)
Comment (jstarpl) [reply meteor/client/ui/RundownView.tsx:1506]: cast to `Pick<RundownPlaylist, '_id' | 'studioId'> | undefined`.

File: meteor/client/lib/MeteorReactComponent.ts (hunk 2 above)
Comment (jstarpl) [reply meteor/client/lib/MeteorReactComponent.ts:52]: Renamed to `subscriptions`.

File: meteor/client/lib/rundown.ts [hunk 4]
```diff
@@ -281,9 +282,18 @@ export namespace RundownUtils {
 				segmentId: segment._id,
 			}
 		)
-		const activePartInstancesMap = playlist.getActivePartInstancesMap({
-			segmentId: segment._id,
-		})
+		const activePartInstancesMap = playlist.getActivePartInstancesMap(
```
[the remark this answers is not in the document]
Comment (jstarpl) [reply meteor/client/lib/rundown.ts:286]: Cast to a map of `PartInstanceLimited = Omit<PartInstance, 'isTaken' | 'previousPartEndState' | 'takeCount'>`

**Pull request merged**